server/daemon/src/main.rs

```diff
@@ -523,36 +530,41 @@ async fn main() -> ExitCode {
 
                     debug!("{sopt:?}");
 
-                    let healthcheck_url = format!("https://{}/status", config.address);
+
+                    let healthcheck_url = match &sopt.check_origin {
+                        true => format!("{}/status", config.origin),
+                        false => format!("https://{}/status", config.address),
+                    };
+
                     debug!("Checking {healthcheck_url}");
 
 
 
-                    let client = reqwest::ClientBuilder::new()
+                    let mut client = reqwest::ClientBuilder::new()
                         .danger_accept_invalid_certs(sopt.no_verify_tls)
                         .danger_accept_invalid_hostnames(sopt.no_verify_tls)
                         .https_only(true);
-                    // TODO: work out how to pull the CA from the chain
-                    // client = match config.tls_config {
-                    //     Some(tls_config) => {
-                    //         eprintln!("{:?}", tls_config);
-                    //         let mut buf = Vec::new();
-                    //         File::open(tls_config.chain)
-                    //             .unwrap()
-                    //             .read_to_end(&mut buf)
-                    //             .unwrap();
-                    //         eprintln!("buf: {:?}", buf);
-                    //         match reqwest::Certificate::from_pem(&buf){
-                    //             Ok(cert) => client.add_root_certificate(cert),
-                    //             Err(err) => {
-                    //                 error!("Failed to read TLS chain: {err:?}");
-                    //                 client
-                    //             }
-                    //         }
-
-                    //     },
-                    //     None => client,
-                    // };
+                    client = match &sopt.ca_cert {
```

Comment:
IMO you don't need a ca_cert param. Chain files are a series of pem, so just passing the chain to from_pem should read the first pem only which is the leaf cert.

Reply:
You'd think that... `reqwest` was *not* impressed.

a) it dies when you pass multiple files
b) it won't trust it based on the leafs
c) I can't remember but there was something else

Reply:
Okay I'm pretty against adding ca_cert as an option too here. We already have the ca available via the chain file.

So I think we either:

* Split the chain file to make reqwest happy.
* Don't validate CA's since we are (presumably) going via localhost anyway ...

Reply:
I'll see if I can get the splitting going, I really don't want to remove the validation option if I can avoid it - see the other comment thread.

Reply:
Could you use https://docs.rs/openssl/latest/openssl/ssl/struct.SslContextBuilder.html#method.set_certificate_chain_file somehow?

Reply:
Splittin' works, just grabbed the last one 😄